### PR TITLE
feat: Contact and ContactAddress, warning when setting to longer than…

### PR DESCRIPTION
**Pull request by pkar70**
… UWP

GitHub Issue (If applicable): #

## PR Type

What kind of change does this PR introduce?
- Feature

## What is the current behavior?
Contact and ContactAddress fields can be set to longer than UWP.
Currently, no warning is produced.

## What is the new behavior?
Setting field to longer than UWP len limit produces warning.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [X] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [X] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information
This is part of #2544, waiting for over year now. In the meantime, MartinZikmund implemented these struct/class, but I think such "length warning" is nice feature.
As I'm recompiling my apps with newer Uno, and rebasing my PRs, some of them reports conflicts to be resolved. This PR is such "change merging".

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UWP/ApplicationModel/Contacts/Contact.cs
Comment (jeromelaban) [inline src/Uno.UWP/ApplicationModel/Contacts/Contact.cs:78]: This may fail is _firstName is null, same for the other properties. Add `#nullable enable` at the top of the file.

This also repetitive throughout the file, make a method that validates this instead e.g. `ValidateLength("Windows.ApplicationModel.Contacts.Contact.FirstName", 64)`

File: src/Uno.UWP/ApplicationModel/Contacts/ContactAddress.cs
Comment (jeromelaban) [inline src/Uno.UWP/ApplicationModel/Contacts/ContactAddress.cs:5]: Place this at the top of the file.

File: src/Uno.UWP/ApplicationModel/Contacts/ContactAddress.cs
Comment (jeromelaban) [inline src/Uno.UWP/ApplicationModel/Contacts/ContactAddress.cs:29]: Use another method for the validation for all properties.

File: src/Uno.UWP/ApplicationModel/Contacts/Contact.cs
Comment (pkar70) [reply src/Uno.UWP/ApplicationModel/Contacts/Contact.cs:78]: a) how _firstName can be null? It is initialized as "".
b) I don't know how implement such ValidateLength method (with property name).

**Comment by pkar70** [comment]
Logging moved to private method. But checks are done inside setter - it is more readable than version with checks inside logging method.

File: src/Uno.UWP/ApplicationModel/Contacts/Contact.cs
Comment (jeromelaban) [inline src/Uno.UWP/ApplicationModel/Contacts/Contact.cs:26]: Use string interpolation instead.

File: src/Uno.UWP/ApplicationModel/Contacts/Contact.cs
Comment (jeromelaban) [inline src/Uno.UWP/ApplicationModel/Contacts/Contact.cs:22]: It's best to use full words:

```suggestion
		private void ValidateStringLength(string value, int maxLength, string variableName)

```

File: src/Uno.UWP/ApplicationModel/Contacts/Contact.cs
Comment (jeromelaban) [inline src/Uno.UWP/ApplicationModel/Contacts/Contact.cs:88]: Move the comparison in the validation method

File: src/Uno.UWP/ApplicationModel/Contacts/ContactAddress.cs
Comment (jeromelaban) [inline src/Uno.UWP/ApplicationModel/Contacts/ContactAddress.cs:21]: Use string interpolation

File: src/Uno.UWP/ApplicationModel/Contacts/ContactAddress.cs
Comment (jeromelaban) [inline src/Uno.UWP/ApplicationModel/Contacts/ContactAddress.cs:17]: Same name change as above.

**Comment by pkar70** [comment]
Ok. Although I think such change make code less readable.

**Comment by davidjohnoliver** [comment]
> chore: empty change - forcing build restart

@pkar70 Please don't push empty commits. Once the PR is approved and ready to merge, the core team will make sure the build passes and it gets merged.

**Comment by pkar70** [comment]
My PC is 14 years old (Core 2 Duo E6550). I'm unable to build full Uno, only "android-only". So, I can't catch any problems with diffignore or similar. Even switching branches takes between 5 and 10 minutes (somewhat quicker is to quit VStudio, change branch from command line, and start VStudio again).
Also, it seems that core team doesn't react for PRs before it passes tests :)

File: src/Uno.UWP/ApplicationModel/Contacts/ContactAddress.cs
Comment (MartinZikmund) [inline src/Uno.UWP/ApplicationModel/Contacts/ContactAddress.cs:7]: Remove this please

File: src/Uno.UWP/ApplicationModel/Contacts/ContactAddress.cs
Comment (MartinZikmund) [inline src/Uno.UWP/ApplicationModel/Contacts/ContactAddress.cs:18]: Move this below all other content (private methods come last in order)

File: src/Uno.UWP/ApplicationModel/Contacts/Contact.cs
Comment (MartinZikmund) [inline src/Uno.UWP/ApplicationModel/Contacts/Contact.cs:22]: Similar here - please move the method below all other members, as private methods should be last in order

**Review by MartinZikmund — APPROVED** [review]

**Comment by pkar70** [comment]
Can this PR get second approval, and "ready to merge" label? It has no real code...

**Review by jeromelaban — APPROVED** [review]

**Pull request merged**
